### DISCUSSION TITLE
Difference between "externalIp" and "externalEndpoint" in the Service entity

### DISCUSSION THREAD

**Discussion by Sgitario** [halkyonio/primaza-poc#230] [Q&A]
The "externalIp" property was added in https://github.com/halkyonio/primaza-poc/pull/229 and is populated from the Ingress resource.
The "externalEndpoint" property was added in https://github.com/halkyonio/primaza-poc/pull/187 and is populated by the user when creating a new service. However, this property is not used anywhere yet. 

Questions:
- Should we rename these two properties to avoid misunderstanding?
- Is correct that the property "externalEndpoint" is there, but not finished yet? If so, is there an issue to complete this functionality? I guess that this property is meant to cope with the scenario where a service is deployed in standalone mode.

**Comment by aureamunoz** [comment]
I think there is another discussion about `externalEndpoint`, [this one.](https://github.com/halkyonio/primaza-poc/discussions/135).
`Is correct that the property "externalEndpoint" is there, but not finished yet?`--> yes. It's never used.
`is there an issue to complete this functionality?`--> I don't see anything about it

I think we should,if possible, merge both properties, and add the logic needed to populate it properly: 

- When registering a service that user knows is running in a standalone mode, user must to full fill the endpoint to access to it.
- The `checkCluster` and `linkServiceInCluster` methods should evolve to check/link services not running in cluster (is this needed? As we should have an IP, the place where service is running should be transparent, no?
- In `generateUrlByClaimService`method case 3&4 could be merged also?

WDYT? 
To be investigated and create an issue in consequence.

**Reply by Sgitario** [reply]
I think the items you describe make sense to be done. 
I'm converting this discussion into an issue to implement it.
